### ISSUE TITLE
About SGHMC

### ISSUE THREAD

**Issue by ALEX5874** [realcrane/3D-student-splatting-and-scooping#2]
Thanks for your amazing work!
I wonder how to tune the parameter related to SGHMC. I found it failed to work when I utilize it in 3DGS-MCMC and use the parameter in your config.
Could you provide some details of how to tune the parameter of SGHMC?

**Comment by Misaliet** [comment]
Hi,


Compared to SGLD in 3DGS-MCMC, SGHMC is a second-order optimizer, so the parameters are not completely universal (although in theory you can use the parameters of one to infer the parameters of the other). I think there are two important points for parameter tuning:

1. The role of SGLD and SGHMC is to introduce noise to allow the optimizer to find a better local minimum. For SGLD, the noise added during optimization is in formula (7) in its paper, and the specific form of the noise is in formula (8). For SGHMC, the more detailed part of the corresponding noise is in formulas (40) and (41) in the Supplementary Material of our paper. For ease of understanding, I omitted the `sigmoid` function used to filter dead components. We added a burn-in stage to SGHMC, which removes the momentum-related values ​​of SGHMC and leaves only pure noise (actually a SGLD). For SGHMC in the burn-in stage, it is the same as SGLD. If the lambda_noise value of SGLD is 5e5, then its total noise is `5e5 * Normal`. If we use the same value to assign `C_burnin`. According to the formula (41) in our Supplementary Material, the real lambda value of our noise is `2*learning_rate^(3/2)*C_burin*Normal`. After burn in, our total noise is `learning_rate*(1-learning_rate*C)*momentum + 2*learning_rate^(3/2)*C_burin*N`. If you want SGLD and SGHMC to have similar effects, you only need to make sure that the values ​​of `5e5 * Normal` and `learning_rate*(1-learning_rate*C)*momentum + 2*learning_rate^(3/2)*C_burin*N` or `(2*learning_rate^(3/2)*C_burin*Normal)` are roughly equal.

2. 3DGS will adjust the actual learning rate according to the size of each scene by calculating a learning rate extent value based on the position of the camera (see the data loader code). The actual learning rate is `learning_rate * extent`. This is also applicable to SGLD. However, if you look at formulas (40) and (41) in our Supplementary Material, you will find that the square of the learning_rate is used in the gradient calculation in SGHMC. So if we follow the default processing of 3DGS, our actual learning rate will become `(learning_rate*extent)^2`, which is obviously a problem. So I processed the learning rate to fit SGHMC, and you can see the code in the `train.py` script.

These are the two most critical parts in my opinion.
